File: src/bindings/ruby/test/TestInitialAssignment.rb
#
# @file    TestInitialAssignment.rb
# @brief   SBML InitialAssignment unit tests
# @author  Akiya Jouraku (Ruby conversion)
# @author  Sarah Keating 
#
# $Id$
# $Source$
#
# This test file was converted from src/sbml/test/TestInitialAssignment.c
# wiht the help of conversion sciprt (ctest_converter.pl).
#
#<!---------------------------------------------------------------------------
# This file is part of libSBML.  Please visit http://sbml.org for more
# information about SBML, and the latest version of libSBML.
#
# Copyright 2005-2008 California Institute of Technology.
# Copyright 2002-2005 California Institute of Technology and
#                     Japan Science and Technology Corporation.
# 
# This library is free software; you can redistribute it and/or modify it
# under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation.  A copy of the license agreement is provided
# in the file named "LICENSE.txt" included with this software distribution
# and also available online as http://sbml.org/software/libsbml/license.html
#--------------------------------------------------------------------------->*/
#
require 'test/unit'
require 'libSBML'

class TestInitialAssignment < Test::Unit::TestCase

  def setup
    @@ia = LibSBML::InitialAssignment.new
    if (@@ia == nil)
    end
  end

  def test_InitialAssignment_create
    assert( @@ia.getTypeCode == LibSBML::SBML_INITIAL_ASSIGNMENT )
    assert( @@ia.getMetaId == "" )
    assert( @@ia.getNotes == nil )
    assert( @@ia.getAnnotation == nil )
    assert( @@ia.getSymbol == "" )
    assert( @@ia.getMath == nil )
  end

  def test_InitialAssignment_createWith
    ia = LibSBML::InitialAssignment.new("k")
    assert( ia.getTypeCode == LibSBML::SBML_INITIAL_ASSIGNMENT )
    assert( ia.getMetaId == "" )
    assert( ia.getNotes == nil )
    assert( ia.getAnnotation == nil )
    assert_equal false, ia.isSetMath
    assert ((  "k" == ia.getSymbol ))
    assert_equal true, ia.isSetSymbol
  end

  def test_InitialAssignment_free_NULL
  end

  def test_InitialAssignment_setMath
    math = LibSBML::parseFormula("2 * k")
    @@ia.setMath(math)
    math1 = @@ia.getMath
    assert( math1 != nil )
    formula = LibSBML::formulaToString(math1)
    assert( formula != nil )
    assert ((  "2 * k" == formula ))
    assert( @@ia.getMath != math )
    assert_equal true, @@ia.isSetMath
    @@ia.setMath(@@ia.getMath)
    math1 = @@ia.getMath
    assert( math1 != nil )
    formula = LibSBML::formulaToString(math1)
    assert( formula != nil )
    assert ((  "2 * k" == formula ))
    assert( @@ia.getMath != math )
    @@ia.setMath(nil)
    assert_equal false, @@ia.isSetMath
    if (@@ia.getMath != nil)
    end
  end

  def test_InitialAssignment_setSymbol
    symbol = "k2"
    @@ia.setSymbol(symbol)
    assert (( symbol == @@ia.getSymbol ))
    assert_equal true, @@ia.isSetSymbol
    if (@@ia.getSymbol == symbol)
    end
    @@ia.setSymbol(@@ia.getSymbol)
    assert (( symbol == @@ia.getSymbol ))
    @@ia.setSymbol("")
    assert_equal false, @@ia.isSetSymbol
    if (@@ia.getSymbol != nil)
    end
  end

end

File: src/bindings/csharp/test/TestInitialAssignment.cs
/// 
///  @file    TestInitialAssignment.cs
///  @brief   SBML InitialAssignment unit tests
///  @author  Frank Bergmann (Csharp conversion)
///  @author  Akiya Jouraku (Csharp conversion)
///  @author  Sarah Keating 
/// 
///  $Id:$
///  $HeadURL:$
/// 
///  This test file was converted from src/sbml/test/TestInitialAssignment.c
///  with the help of conversion sciprt (ctest_converter.pl).
/// 
/// <!---------------------------------------------------------------------------
///  This file is part of libSBML.  Please visit http://sbml.org for more
///  information about SBML, and the latest version of libSBML.
/// 
///  Copyright 2005-2008 California Institute of Technology.
///  Copyright 2002-2005 California Institute of Technology and
///                      Japan Science and Technology Corporation.
///  
///  This library is free software; you can redistribute it and/or modify it
///  under the terms of the GNU Lesser General Public License as published by
///  the Free Software Foundation.  A copy of the license agreement is provided
///  in the file named "LICENSE.txt" included with this software distribution
///  and also available online as http://sbml.org/software/libsbml/license.html
/// --------------------------------------------------------------------------->*/


namespace LibSBMLCSTest {

  using libsbml;

  using  System.IO;

  public class TestInitialAssignment {
    public class AssertionError : System.Exception 
    {
      public AssertionError() : base()
      {
        
      }
    }


    static void assertTrue(bool condition)
    {
      if (condition == true)
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        return;
      }
      else if (a.Equals(b))
      {
        return;
      }
  
      throw new AssertionError();
    }

    static void assertNotEquals(object a, object b)
    {
      if ( (a == null) && (b == null) )
      {
        throw new AssertionError();
      }
      else if (a.Equals(b))
      {
        throw new AssertionError();
      }
    }

    static void assertEquals(bool a, bool b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(bool a, bool b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertEquals(int a, int b)
    {
      if ( a == b )
      {
        return;
      }
      throw new AssertionError();
    }

    static void assertNotEquals(int a, int b)
    {
      if ( a != b )
      {
        return;
      }
      throw new AssertionError();
    }

    private InitialAssignment IA;

    public void setUp()
    {
      IA = new  InitialAssignment();
      if (IA == null);
      {
      }
    }

    public void tearDown()
    {
      IA = null;
    }

    public void test_InitialAssignment_create()
    {
      assertTrue( IA.getTypeCode() == libsbml.SBML_INITIAL_ASSIGNMENT );
      assertTrue( IA.getMetaId()== "" == true );
      assertTrue( IA.getNotes() == null );
      assertTrue( IA.getAnnotation() == null );
      assertTrue( IA.getSymbol()== "" == true );
      assertTrue( IA.getMath() == null );
    }

    public void test_InitialAssignment_createWith()
    {
      InitialAssignment ia = new  InitialAssignment("k");
      assertTrue( ia.getTypeCode() == libsbml.SBML_INITIAL_ASSIGNMENT );
      assertTrue( ia.getMetaId()== "" == true );
      assertTrue( ia.getNotes() == null );
      assertTrue( ia.getAnnotation() == null );
      assertEquals( false, ia.isSetMath() );
      assertTrue((  "k" == ia.getSymbol() ));
      assertEquals( true, ia.isSetSymbol() );
      ia = null;
    }

    public void test_InitialAssignment_free_NULL()
    {
    }

    public void test_InitialAssignment_setMath()
    {
      ASTNode math = libsbml.parseFormula("2 * k");
      string formula;
      ASTNode math1;
      IA.setMath(math);
      math1 = IA.getMath();
      assertTrue( math1 != null );
      formula = libsbml.formulaToString(math1);
      assertTrue( formula != null );
      assertTrue((  "2 * k" == formula ));
      assertTrue( IA.getMath() != math );
      assertEquals( true, IA.isSetMath() );
      IA.setMath(IA.getMath());
      math1 = IA.getMath();
      assertTrue( math1 != null );
      formula = libsbml.formulaToString(math1);
      assertTrue( formula != null );
      assertTrue((  "2 * k" == formula ));
      assertTrue( IA.getMath() != math );
      IA.setMath(null);
      assertEquals( false, IA.isSetMath() );
      if (IA.getMath() != null);
      {
      }
      math = null;
    }

    public void test_InitialAssignment_setSymbol()
    {
      string Symbol = "k2";
      IA.setSymbol(Symbol);
      assertTrue(( Symbol == IA.getSymbol() ));
      assertEquals( true, IA.isSetSymbol() );
      if (IA.getSymbol() == Symbol);
      {
      }
      IA.setSymbol(IA.getSymbol());
      assertTrue(( Symbol == IA.getSymbol() ));
      IA.setSymbol("");
      assertEquals( false, IA.isSetSymbol() );
      if (IA.getSymbol() != null);
      {
      }
    }

  }
}
